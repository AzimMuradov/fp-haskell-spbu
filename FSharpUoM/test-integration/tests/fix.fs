let rec fix f = f (fix f);;

let rec fix f x = f (fix f) x; 
fix (fun re n -> if n <= 1 then 1 else n * re (n - 1));;

let rec fix f x = f (fix f) x; 
fix (fun re n -> if n <= 1 then 1 else n * re (n - 1)) 10;;
