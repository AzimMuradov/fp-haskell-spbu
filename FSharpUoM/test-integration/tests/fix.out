val fix: forall a6. (a6 -> a6) -> a6
val it: int -> int
val it: int = 3628800
